src/metabase/api/dashboard.clj

```diff
@@ -1129,6 +1129,9 @@
     (keyword (name type))
     :=))
 
+;; TODO needs to call [[lib/ensure-filter-stage]] and take `stage-number` from the parameter mapping into account
+;; TODO duplicates code in params.clj
+;; TODO needs to wrap models and metrics properly!
```

Comment:
still TODO?

Reply:
@bshepherdson this still needs to be fixed. It's not only that the logic is duplicated - it doesn't handle cases like models, metrics, post-aggregation filtering